src/wallet/storage.rs

```diff
@@ -35,6 +35,8 @@ pub(crate) struct WalletStore {
     /// Map of prevout to contract redeemscript.
     pub(super) prevout_to_contract_map: HashMap<OutPoint, ScriptBuf>,
     /// Map for all the fidelity bond information. (index, (Bond, script_pubkey, is_spent)).
+    /// TODO: The hashmap doesn't require to script-pubkey as a additional parameter -> we can directly use `script_pub_key` api wherever required.
+    /// TODO: Can we also remove  `is_spent` inside the fidelity bond struct?
```

Comment:
let's remove it in a followup PR. I think its redundant. If we ever need it we can add it back.